core/arch/arm/kernel/virtualization.c

```diff
@@ -0,0 +1,51 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/* Copyright (c) 2018 EPAM Systems. All rights reserved. */
+
+#include <compiler.h>
+#include <platform_config.h>
+#include <kernel/panic.h>
+#include <kernel/virtualization.h>
+#include <kernel/spinlock.h>
+#include <sm/optee_smc.h>
+
+#define INVALID_CLIENT_ID		0xFFFF
+
+static uint16_t current_client_id = 0;
```

Comment:
minor: `0` is a specific ID for the hypervisor, right?
Maybe init `current_client_id` to `INVALID_CLIENT_ID`. Or discard need for `INVALID_CLIENT_ID`.